pkg/controller/istio/istio_controller.go

```diff
@@ -0,0 +1,348 @@
+		},
+		"ambient": map[string]interface{}{
+			"enabled":                    true,
+			"reconcileIptablesOnStartup": true,
+		},
+	}
+	if installation.KubernetesProvider == operatorv1.ProviderGKE {
+		cniOpts["global"].(map[string]interface{})["platform"] = "gke"
+	}
+	ztunnelOpts := map[string]interface{}{
+		"image": istioComponent.IstioZtunnelImage,
+		"global": map[string]interface{}{
+			"istioNamespace": render.IstioNamespace,
+		},
+	}
+	istioCfg.Resources, err = renderistio.GetResources(render.IstioNamespace, render.IstioReleaseName, baseOpts,
+		istiodOpts, cniOpts, ztunnelOpts)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceCreateError, "Error rendering Istio resources", err, log)
+		return reconcile.Result{}, err
+	}
+
+	// Deploy Istio components
+	err = utils.NewComponentHandler(log, r, r.scheme, istio).CreateOrUpdateOrDelete(ctx, istioComponent, r.status)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceCreateError, "Error rendering Istio resources", err, log)
+		return reconcile.Result{}, err
+	}
+
+	// Check all components are ready
+	istiodDep := &appsv1.Deployment{}
+	if err = r.Get(ctx, client.ObjectKey{Namespace: render.IstioNamespace,
+		Name: render.IstioIstiodDeploymentName}, istiodDep); err != nil {
+
+		r.status.SetDegraded(operatorv1.ResourceNotFound, "Istiod deployment not found", err, log)
+		return reconcile.Result{}, err
+	}
+	if istiodDep.Spec.Replicas == nil || istiodDep.Status.ReadyReplicas != *istiodDep.Spec.Replicas {
+		r.status.SetDegraded(operatorv1.ResourceNotReady, "Istiod deployment not ready", nil, log)
+		return reconcile.Result{RequeueAfter: time.Second * 5}, nil
+	}
+
+	istioCniDs := &appsv1.DaemonSet{}
+	if err = r.Get(ctx, client.ObjectKey{Namespace: render.IstioNamespace,
+		Name: render.IstioCNIDaemonSetName}, istioCniDs); err != nil {
+
+		r.status.SetDegraded(operatorv1.ResourceNotFound, "Istio CNI daemonset not found", err, log)
+		return reconcile.Result{}, err
+	}
+	if istioCniDs.Status.NumberReady != istioCniDs.Status.DesiredNumberScheduled {
+		r.status.SetDegraded(operatorv1.ResourceNotReady, "Istio CNI daemonset not ready", nil, log)
+		return reconcile.Result{RequeueAfter: time.Second * 5}, nil
+	}
+
+	ztunnelDs := &appsv1.DaemonSet{}
+	if err = r.Get(ctx, client.ObjectKey{Namespace: render.IstioNamespace,
+		Name: render.IstioZTunnelDaemonSetName}, ztunnelDs); err != nil {
+
+		r.status.SetDegraded(operatorv1.ResourceNotFound, "ZTunnel daemonset not found", err, log)
+		return reconcile.Result{}, err
+	}
+	if ztunnelDs.Status.NumberReady != ztunnelDs.Status.DesiredNumberScheduled {
+		r.status.SetDegraded(operatorv1.ResourceNotReady, "ZTunnel daemonset not ready", nil, log)
+		return reconcile.Result{RequeueAfter: time.Second * 5}, nil
+	}
+
+	// Clear the degraded bit if we've reached this far.
+	r.status.ClearDegraded()
+
+	// Update the status of the Istio instance and StatusManager.
+	return reconcile.Result{RequeueAfter: utils.PeriodicReconcileTime}, nil
```

Comment:
You should remove this RequeueAfter and instead use the periodic reconcile source, like here: https://github.com/tigera/operator/blob/2f00b572bc2acc2b43ad3da94ddbdcc98e997db3/pkg/controller/gatewayapi/gatewayapi_controller.go#L92-L96